### ISSUE TITLE
Rendering changes with multibyte prompt characters in recent versions.

### ISSUE THREAD

**Issue by xwmx** [gwsw/less#576]
The rendering of multibyte characters in the prompt appears to have changed in recent versions and the characters are no longer rendered properly on my system.

Given the following command:

```bash
echo "Example content." | less --prompt "Example prompt ↓↑"
```

`less` v653 and earlier render the prompt as expected:

<img width="596" alt="screenshot 1" src="https://github.com/user-attachments/assets/4b52d5b7-fc2b-4346-9e41-bac6e3632790">

In v654, `less` started printing the code points in angle brackets with leading `^@` characters:

<img width="596" alt="screenshot 2" src="https://github.com/user-attachments/assets/b12a3d5f-6820-4a60-bc15-20987dada72c">

At some point this changed so the leading `^@` is now omitted in the current release version, v667, and the master branch, which currently print the code points in angle brackets:

<img width="596" alt="screenshot 3" src="https://github.com/user-attachments/assets/b156e898-a1d7-4ccf-973c-a31254ebaf21">

When `$LESSCHARSET` is set to `dos`, the characters render properly, as illustrated in the following comparison:

```bash
echo "Example content." | LESSCHARSET=utf-8 less --prompt "Example prompt ↓↑"
# Example content.
# Example prompt <U+0086><U+0093><U+0086><U+0091>

echo "Example content." | LESSCHARSET=dos less --prompt "Example prompt ↓↑"
# Example content.
# Example prompt ↓↑
```

The system is running macOS 14.6.1 and the output of `locale` is:

```bash
LANG="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_CTYPE="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_ALL="en_US.UTF-8"
```

Thank you for developing such a useful and important program!

**Comment by gwsw** [comment]
Should be fixed in 5a369eda701941e239bef002d624f37528bf8e4b.

**Comment by avih** [comment]
> Should be fixed in [5a369ed](https://github.com/gwsw/less/commit/5a369eda701941e239bef002d624f37528bf8e4b).

Just FYI that this also fixes the issue on Windows UCRT build, because while commit ad440b6 mentions "unicode file names", it actually enables any unicode CLI arguments, such as the arrows given in this example with `--prompt` (and supposedly env vars too), while the MSVCRT build doesn't show the arrows at all - they probably fail to get translated to the OEM codepage which is the ANSI encoding of argv.

And while on this subject, I wonder if on Windows it would be simpler to use the wide version `wmain` instead of relying on `setlocale` in a UCRT build, so that unicode argv works correctly in all Windows versions (at least XP+) and regardless of the CRT. I might look into that.

**Comment by xwmx** [comment]
@gwsw It's working for me now. Thank you!
